guides/micronaut-microservices-services-discover-eureka/bookrecommendation/groovy/src/main/groovy/example/micronaut/Book.groovy

```diff
@@ -2,14 +2,29 @@ package example.micronaut
 
 import groovy.transform.CompileStatic
 import groovy.transform.EqualsAndHashCode
-import groovy.transform.TupleConstructor
+import io.micronaut.core.annotation.Creator
 import io.micronaut.core.annotation.Introspected
+import io.micronaut.core.annotation.NonNull
+
+import javax.validation.constraints.NotBlank
 
 @CompileStatic
-@TupleConstructor
 @EqualsAndHashCode
 @Introspected
 class Book {
-    String isbn
-    String name
+
+    @NonNull
+    @NotBlank
+    final String isbn
+
+    @NonNull
+    @NotBlank
+    final String name
+
+    @Creator
```

Comment:
I don't think `@Creator` is necessary. Am I missing anything?

If you want to specify it to educate the user about `@Creator` I think we should add a callout with a callout snippet.